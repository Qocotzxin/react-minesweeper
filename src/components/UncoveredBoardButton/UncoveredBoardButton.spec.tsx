import { render, screen } from "@testing-library/react";
import { UncoveredBoardButton } from "./UncoveredBoardButton";

describe("UncoveredBoardButton", () => {
  describe("Snapshot", () => {
    it("Should render correctly and match snapshot.", () => {
      const { container } = render(<UncoveredBoardButton />);

      expect(container).toMatchSnapshot();
    });
  });

  describe("Behavior", () => {
    it("Should render render children as it is when its not equal to *.", () => {
      render(<UncoveredBoardButton>1</UncoveredBoardButton>);

      expect(
        screen.queryByTestId("UncoveredBoardButton-bomb-icon")
      ).not.toBeInTheDocument();
    });

    it("Should render a bomb icon when children passed is *.", () => {
      render(<UncoveredBoardButton>*</UncoveredBoardButton>);

      expect(
        screen.getByTestId("UncoveredBoardButton-bomb-icon")
      ).toBeInTheDocument();
    });
  });
});
