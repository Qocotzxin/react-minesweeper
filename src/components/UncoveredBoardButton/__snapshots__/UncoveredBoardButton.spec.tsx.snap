// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UncoveredBoardButton Snapshot Should render correctly and match snapshot. 1`] = `
<div>
  <div
    class="MuiBox-root css-1mbwnhf"
    data-testid="UncoveredBoardButton"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-vhbbj2-MuiTypography-root"
    />
  </div>
</div>
`;
